ArduSub/system.cpp

```diff
@@ -49,7 +49,7 @@ void Sub::init_ardupilot()
         AP_Param::set_default_by_name("BARO_EXT_BUS", 1);
         break;
     }
-#else
+#elif CONFIG_HAL_BOARD != HAL_BOARD_LINUX
```

Comment:
These sub-specific patches for board configuration really need to die.

Do we have a way forward for that?

Reply:
I agree they should die, but we don't have a plan for it. I'll get a Pixhawk4 soon, then maybe I'll be able to find a better solution that works across more boards.